cmd/skaffold/app/cmd/dev.go

```diff
@@ -85,7 +85,10 @@ func devLoop(ctx context.Context, out io.Writer, filename string) error {
 				ctxDev, cancelDev := context.WithCancel(ctx)
 				c <- cancelDev
 				if err := runDev(ctxDev, out, filename); err != nil {
-					logrus.Warnln("dev:", err)
+					logrus.Errorln("dev:", err)
+					cancelMainLoop()
```

Comment:
I think this is getting deferred before we enter the dev loop, does it matter that the `cancel` function might get called twice?